### PR TITLE
Add prefix 'Abstract' in the name for all the relevant abstract classes

### PR DESCRIPTION
**Pull request by kiran-vadla**
Fix for Issue #561 is provided as part of this PR . As a fix . added prefix 'Abstract' in the name for all the relevant abstract classes.

Since the changes as part of the review comments of PR #576 were many , this fresh PR is raised . PR #576 will be closed . 

Thanks,
@kiran-vadla

### CODE REVIEW

File: modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControl.java [hunk 1]
```diff
@@ -1,100 +1,17 @@
...
-
-    super();
-    this.id = id;
-  }
+public interface AccessControl {
 
   /**
-   * @return the unique identifier of this {@link AccessControl}. Has to be unique for all {@link AccessControl} in a
-   *         {@link AccessControlSchema}.
+   * @return the unique identifier of {@link AbstractAccessControl}. Has to be unique for all
```
Comment (amarinso) [inline modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControl.java]: You have to check the comments. Almost all references in comments to `AbstractAccessControl` have to be really to `AccessControl` instead

File: modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AccessControlBasedAuthenticationProvider.java [hunk 2]
```diff
@@ -23,31 +23,31 @@
...
- * @param
- *        <P>
- *        is the generic type of the {@link Principal} for internal user representation to bridge with
+ * @param <P> is the generic type of the {@link Principal} for internal user representation to bridge with
  *        {@link PrincipalAccessControlProvider}.
  *
  */
 @Deprecated
-public abstract class AbstractAccessControlBasedAuthenticationProvider<U extends UserDetails, P extends Principal>
+public abstract class AccessControlBasedAuthenticationProvider<U extends UserDetails, P extends Principal>
```
Comment (amarinso) [inline modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AccessControlBasedAuthenticationProvider.java]: Why is this no longer *abstract* ?

Also, it says deprecated on the previous version... is it still being used? if not, we should just delete it (And document in the migration page)

File: modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControl.java (hunk 1 above)
Comment (kiran-vadla) [inline modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControl.java]: @amarinso , Please check now.

File: modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AccessControlBasedAuthenticationProvider.java (hunk 2 above)
Comment (kiran-vadla) [inline modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AccessControlBasedAuthenticationProvider.java]: @amarinso , Please check now . These are deleted .

**Comment by kiran-vadla** [comment]
@amarinso : All references are changed to AccessControl from AbstractAccessControl in the comments section of different classes.  Please review . 

Thanks,
@kiran-vadla

**Pull request merged**